pkg/integration/tests/stash/stash.go

```diff
@@ -0,0 +1,27 @@
+package stash
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var Stash = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Stashing files",
+	ExtraCmdArgs: "",
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
+	SetupRepo: func(shell *Shell) {
+		shell.EmptyCommit("initial commit")
+		shell.CreateFile("file", "content")
+		shell.GitAddAll()
+	},
+	Run: func(shell *Shell, input *Input, assert *Assert, keys config.KeybindingConfig) {
```

Comment:
you mentioned that there were no stashing examples in the existing tests. Although that's true, there are a few examples that can help with writing a test in general.

Because this test will press keys as fast as possible, it's up to you to have assertions after each key press to ensure the expected effect has occurred. So for example, when you go to view the stash options you can then do:
```go
		assert.InMenu()
		assert.MatchCurrentViewTitle(Equals("foo"))
		assert.MatchSelectedLine(Contains("bar"))
		
		input.Confirm()
		
		assert.InConfirm()
		assert.MatchCurrentViewTitle(Equals("Are you sure?"))
		input.Confirm()
```

Reply:
Can you tell why this test is failing?

```go
package stash

import (
	"github.com/jesseduffield/lazygit/pkg/config"
	. "github.com/jesseduffield/lazygit/pkg/integration/components"
)

var Stash = NewIntegrationTest(NewIntegrationTestArgs{
	Description:  "Stashing files",
	ExtraCmdArgs: "",
	Skip:         false,
	SetupConfig:  func(config *config.AppConfig) {},
	SetupRepo: func(shell *Shell) {
		shell.EmptyCommit("initial commit")
		shell.CreateFile("file", "content")
		shell.GitAddAll()
	},
	Run: func(shell *Shell, input *Input, assert *Assert, keys config.KeybindingConfig) {
		assert.StashCount(0)

		input.PressKeys(keys.Files.ViewStashOptions)
		assert.InMenu()

		input.PressKeys("a")
		input.Confirm()

		assert.StashCount(1)
	},
})

```

Here's the `StashCount` function:

```go
func (self *Assert) StashCount(expectedCount int) {
	self.assertWithRetries(func() (bool, string) {
		actualCount := len(self.gui.Model().StashEntries)

		return actualCount == expectedCount, fmt.Sprintf(
			"Expected %d stash entries, but got %d",
			expectedCount, actualCount,
		)
	})
}
```

I ran this command:
```
go run cmd/integration_test/main.go cli --slow stash/stash
```

Reply:
> ```
> 	Run: func(shell *Shell, input *Input, assert *Assert, keys config.KeybindingConfig) {
>		assert.StashCount(0)
>
>		input.PressKeys(keys.Files.ViewStashOptions)
>		input.Confirm()
>		// input.Type("stash name")
>		input.Confirm()
>
>		assert.StashCount(1)
>
> 	},
> ```

This is from the checkout, not from your post, it differs for some reason.

Anyway, since you don't give the stash a name, it takes the name in the form of `WIP on master: <hash> initial commit`, and since the `hash` is based on several [timestamps](https://stackoverflow.com/a/23792149), the stash name changes with each test run.

The solution is to give the `stash` a name, any name will suffice.